src/App.jsx

```diff
@@ -14,33 +16,100 @@
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD__ALPHABET = 'alphabet';
+const SORT__FIELD__LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoods(goods, { sortField, reverse }) {
+  let preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD__ALPHABET:
+          return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
+        case SORT__FIELD__LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    reverse,
+  });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(SORT_FIELD__ALPHABET)}
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortField !== SORT_FIELD__ALPHABET,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortField(SORT__FIELD__LENGTH)}
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortField !== SORT__FIELD__LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          onClick={() => setReverse(prev => !prev)}
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': !reverse,
+          })}
+        >
+          Reverse
+        </button>
+
+        {(sortField || reverse) && (
+          <button
+            onClick={() => {
+              setReverse(false);
+              setSortField('');
```

Comment:
When resetting the sort field and reverse state, ensure that the logic is clear and that both states are reset to their initial values. The current implementation seems correct, but double-check that this aligns with the intended behavior.